--- codegen.py	2018-08-26 15:34:55.703648328 +0200
+++ codegen_new.py	2018-08-26 15:51:32.775110763 +0200
@@ -498,7 +498,10 @@
     def __init__(self, source_cls, target_cls, assocation_table, inflect_engine):
         super(ManyToManyRelationship, self).__init__(source_cls, target_cls)
 
-        self.kwargs['secondary'] = repr(assocation_table.schema + '.' + assocation_table.name)
+        try:
+            self.kwargs['secondary'] = repr(assocation_table.schema + '.' + assocation_table.name)
+        except:
+            self.kwargs['secondary'] = "'"+str(assocation_table)+"'"
         constraints = [c for c in assocation_table.constraints if isinstance(c, ForeignKeyConstraint)]
         constraints.sort(key=_get_constraint_sort_key)
         colname = _get_column_names(constraints[1])[0]
